pkg/shed/field_struct.go

```
@@ -67,12 +67,15 @@ func (f StructField) Put(val interface{}) (err error) {
 }
 
 // PutInBatch marshals provided val and puts it into the batch.
-func (f StructField) PutInBatch(batch *leveldb.Batch, val interface{}) (err error) {
+func (f StructField) PutInBatch(batch *badger.Txn, val interface{}) (err error) {
 	b, err := json.Marshal(val)
 	if err != nil {
-		f.logger.Debugf("could not PUT key %s in batch", string(f.key))
+		f.logger.Debugf("could not marshall while doing putiInBatch. Error: %s", err.Error())
 		return err
 	}
-	batch.Put(f.key, b)
+	err = batch.Set(f.key, b)
+	if err != nil {
+		f.logger.Debugf("could not put values in batch. Error: %s", err.Error())
```

Why the error is not returned, just logged?

---

Do not log error and return it.